streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -760,23 +762,39 @@ private boolean assignTasksToClients(final Cluster fullMetadata,
         log.debug("Assigning tasks and {} standby replicas to client nodes {}",
                   numStandbyReplicas(), clientStates);
 
-        final TaskAssignor taskAssignor = createTaskAssignor(lagComputationSuccessful);
-
-        final RackAwareTaskAssignor rackAwareTaskAssignor = new RackAwareTaskAssignor(
-            fullMetadata,
-            partitionsForTask,
-            changelogTopics.changelogPartionsForTask(),
-            tasksForTopicGroup,
-            racksForProcessConsumer,
-            internalTopicManager,
-            assignmentConfigs,
-            time
-        );
-        final boolean probingRebalanceNeeded = taskAssignor.assign(clientStates,
-                                                                   allTasks,
-                                                                   statefulTasks,
-                                                                   rackAwareTaskAssignor,
-                                                                   assignmentConfigs);
+        final org.apache.kafka.streams.processor.assignment.TaskAssignor userTaskAssignor =
+            createUserTaskAssignor(lagComputationSuccessful);
+        boolean probingRebalanceNeeded = false;
+        if (userTaskAssignor == null) {
```

Comment:
nit: consider making the if and else separate methods that return `boolean` (whether probing rebalance is necessary) and just assign directly so it can be `final`